File: Concurrency/ThreadPoolExecuter/thread_pool_executer.py
import time
from concurrent.futures import ThreadPoolExecutor

def work(n):
    print(f"Task {n} started")
    time.sleep(5)
    print(f"Task {n} finished")


threadPoolExecuter=ThreadPoolExecutor(max_workers=3, thread_name_prefix="This is my name ")

threadPoolExecuter.submit(work,1)
threadPoolExecuter.submit(work,2)
threadPoolExecuter.submit(work,3)
threadPoolExecuter.shutdown(wait=True)

File: Concurrency/ThreadPoolExecuter/thread_pool_executer.cpp
#include <queue>
#include "../../cpp/header.h"

class ThreadPoolExecuter {
    std::queue<int> q;
    vector<thread> thread_pool;
    public:
    void submit(thread t) {
        thread_pool.emplace_back(t);
    }
    void run() {
        for (auto &t:thread_pool) {
            t.
        }
    }
    void shutdown() {

    }

};

void work() {
    cout<<"Working"<<endl;
}
